docs/api.rst

```diff
@@ -102,7 +102,8 @@ Data Utils
 .. autosummary::
    :toctree: generated/
 
-    create_from_X_y
```

Comment:
that's an API breakage without deprecation

Reply:
@agramfort Is it fine to use mne.utils.deprecated to add a DeprecationWarning? Or is this confusing, since the function will then appear to be a mne and not a braindecode function?

Reply:
i would copy such a util function in the braindecode somewhere.

Reply:
Ok, thanks! I will discuss with @robintibor about where to put it and how to do it.